Game flow follow-up work

rr- opened the pull request:
#### Checklist

- [X] I have read the [coding conventions](https://github.com/LostArtefacts/TR1X/blob/master/CONTRIBUTING.md#coding-conventions)
- [X] I have added a changelog entry about what my pull request accomplishes, or it is an internal change
- [X] I have added a readme entry about my new feature or OG bug fix, or it is a different change

#### Description

- Resolves #2360 
- Fixes some tiny memory leaks in TR1 and TR2
- Moves `"demos"` to their own key in the TR2 gameflow JSON
- Does internal refactors to TR1 in preparation to merge via TRX (very preliminary work)
- Merges `start_cine` TR1 sequence event to `start_game`
- Merges `stop_game` TR1 sequence event to `loop_game` (they were always placed together)
- Merges `loop_cine` TR1 sequence event to `loop_game`
- Makes the `level_num` in the `play_level` TR2 sequence event optional
- Removes camera position sequence events from TR1 as they seem to be no longer required

More refactors to come…

rr- committed:
tr2/game-flow: move demos to their own key

rr- committed:
tr2/game-flow: drop N_ prefix

rr- committed:
tr2/game-flow: fix missing shutdown call

rr- committed:
tr2/music: fix memory leaks

rr- committed:
tr2/main: fix memory leak

rr- committed:
tr2/enum-map: fix memory leak

rr- committed:
game-flow: merge common functions

rr- committed:
tr1/game-flow: use separate sequence type

rr- committed:
tr1/game-flow: split into modules

rr- committed:
tr1/game-flow: use enum-map

rr- committed:
tr1/game-flow: fix story so far exiting early

Resolves #2360.

rr- committed:
tr1/game-flow: remove set cam pos sequence events

rr- committed:
tr2/game-flow: remove GFS_PLAY_DEMO

Unused after refactor.

rr- committed:
tr1/game-flow: merge GFS_START_CINE to GFS_START_GAME

rr- committed:
tr2/game-flow: guess level_num in play_level

rr- committed:
tr1/game-flow: merge GFS_STOP_GAME to GFS_LOOP_GAME

rr- committed:
tr1/game-flow: merge GFS_LOOP_CINE to GFS_LOOP_GAME

rr- committed:
game-flow: use common GFS naming

aredfan commented:
**TR2X issues:**

- [x] The fmv before demo 3 seems to play with these steps: 

1. have fmvs enabled
2. /`play 1` (can be any level or demo)
3. `/demo 3`

- [x] In the final cutscene, the sound of the shotgun is cut off more early when compared to a recent dev snapshot. Load the attached save file and eliminate the last enemy [savegame.zip](https://github.com/user-attachments/files/18518430/savegame.zip)
- [x] This might be related to the prior issue - all the cutscenes are playing the CUT1 soundtrack, albeit with a delay.

rr- committed:
tr1/game-flow: fix crash when starting demo and there are no demos

rr- commented:
@aredfan LMK if the issues persist.

lahm86 reviewed (approved):
LGTM

aredfan commented:
> @aredfan LMK if the issues persist.

Initially the PR build didn't fix my latter 2 issues, so I've removed the audio folder from my TR2 directory and that actually fixed the issues.  I think the audio folder is now given priority over the music folder.  Previously, the music folder is given priority.

The issues with the audio folder preceded this PR.

(Apologies for my mistake.)

rr- commented:
> > @aredfan LMK if the issues persist.
> 
> Initially the PR build didn't fix my latter 2 issues, so I've removed the audio folder from my TR2 directory and that actually fixed the issues. I think the audio folder is now given priority over the music folder. Previously, the music folder is given priority.
> 
> The issues with the audio folder preceded this PR.

Thank you for this. You're absolutely right, that's related to the memory leak fix in this PR which didn't stop on the first matching backend and instead would continue to try all others. This is now fixed.
